#4 MyFlights Screen Design

OrlandoAlPe opened the pull request:
Completed design of My flights screen:

- Created floating action button
- Crated List where the flight items will render
- The list is filled with dummy data
- Pending: 
    - Floating action button doesn't have a function.
    - Rendering list items from firestore data

Flight card component where the user can see their flight's details.

- Created the component and their respective properties to make the component dynamic.

![Screenshot_1618340545](https://user-images.githubusercontent.com/33968113/114609301-41d3e300-9c64-11eb-8bb6-10752f6a6137.png)

OrlandoAlPe committed:
#4 implemented

Completed design of my flights screen and created Flight card component where the user can see their flight's dets.

YilKanda commented on reservacionVuelo/Screens/MyFlights.js:
```diff
@@ -0,0 +1,75 @@
+import Icon from 'react-native-vector-icons/FontAwesome';
+import {FlightCard} from '../components/flightCard';
+
+const dummyData = [
+  {
+    id: 1,
+    departureCity: 'AGS',
+    departureCountry: 'MEXICO',
+    arrivalCity: 'LA',
+    arrivalCountry: 'USA',
+    year: 2021,
+    month: 'April',
+    day: 30,
+    passangers: 2,
+  },
+  {
+    id: 2,
+    departureCity: 'COL',
+    departureCountry: 'MEXICO',
+    arrivalCity: 'NY',
+    arrivalCountry: 'USA',
+    year: 2021,
+    month: 'April',
+    day: 22,
+    passangers: 3,
+  },
+  {
+    id: 3,
+    departureCity: 'MED ',
+    departureCountry: 'CDMX',
+    arrivalCity: 'BRL',
+    arrivalCountry: 'GERMANY',
+    year: 2021,
+    month: 'April',
+    day: 28,
+    passangers: 3,
+  },
+];
```
As a suggestion, I recommend that you put this arrangement in another file, so that you don't overload the MyFlights.js file.

YilKanda commented on reservacionVuelo/Screens/MyFlights.js:
```diff
@@ -0,0 +1,75 @@
+          data={dummyData}
+          keyExtractor={item => item.id}
+          renderItem={({item}) => (
+            <FlightCard
+              depCity={item.departureCity}
+              depCountry={item.departureCountry}
+              arrCity={item.arrivalCity}
+              arrCountry={item.arrivalCountry}
+              year={item.year}
+              month={item.month}
+              day={item.day}
+              passangers={item.passangers}
+            />
```
As a suggestion, I recommend that you put this arrangement in another file, so that you don't overload the MyFlights.js file.

YilKanda reviewed (dismissed):
Just leave a few comments. Some code can be used in other components and thereby decrease the workload for the MyFlights.js file

JacquelineFM commented on reservacionVuelo/Screens/MyFlights.js:
```diff
@@ -0,0 +1,75 @@
+import React, {Component} from 'react';
+import {View, Text, TouchableHighlight, ListView, FlatList} from 'react-native';
```
You can remove the _ListView_ component as you are not using it on the screen 😊

JacquelineFM commented on reservacionVuelo/Screens/MyFlights.js:
```diff
@@ -0,0 +1,75 @@
+import Icon from 'react-native-vector-icons/FontAwesome';
+import {FlightCard} from '../components/flightCard';
+
+const dummyData = [
```
WDYT of having your dummy Data in a file maybe inside a folder to better identify it? I know they won't really be used and it's for testing only, but for the same reason I would recommend having it separate, so when you already have the DB implemented you just delete the file and you will not have to modify this file again.

OrlandoAlPe replied:
It was only temporary, but now that we moved the data, the component looks way cleaner!

JacquelineFM replied:
I'm happy to read that! 😊

JacquelineFM commented on reservacionVuelo/components/flightCard.js:
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import {  Text, View } from 'react-native'
+import Icon from 'react-native-vector-icons/FontAwesome';
+import styles from './styles';
+
```
In this file it was only missing about the indentation and the ESLint.

JacquelineFM commented on reservacionVuelo/components/styles.js:
```diff
@@ -0,0 +1,108 @@
+import {StyleSheet} from 'react-native';
+
+const styles = StyleSheet.create({
+  mainContainer: {
+    paddingHorizontal: 30,
+    flex: 1,
+  },
+  myFlightsContainer: {
+    paddingHorizontal: 30,
+    paddingVertical: 10,
+    flex: 1,
+    justifyContent: 'space-between',
```
🔥 Great job!

JacquelineFM reviewed (changes requested):
One last comment, hehehe, it seems that there are problems in the files:

_reservacionVuelo/Screens/SignUp.js
reservacionVuelo/components/styles.js_

So they don't forget to solve them!

OrlandoAlPe committed:
#4 fix

Created data file and moved our dummy data into the data file.
Implemented EsLint in every file.
Removed Unused components.

OrlandoAlPe committed:
Merge branch 'sand-box' into #4-my-flights-design

JacquelineFM reviewed (approved):
Nice job! 😊
